File: invesdwin-scripting-parent/invesdwin-scripting-rust/invesdwin-scripting-rust-runtime-contract/src/test/java/de/invesdwin/scripting/rust/runtime/contract/callback/ParametersAndReturnsTestBoolean.rs
println!("getBoolean");
let getBoolean: bool = callback("getBoolean", &[]);
println!("getBoolean: {}", getBoolean);
callback_void("setBoolean", &[param(getBoolean)]);

println!("getBooleanVector");
let getBooleanVector: Vec<bool> = callback("getBooleanVector", &[]);
println!("getBooleanVector: {:?}", getBooleanVector);
callback_void("setBooleanVector", &[param(getBooleanVector)]);

println!("getBooleanVectorAsList");
let getBooleanVectorAsList: Vec<bool> = callback("getBooleanVectorAsList", &[]);
println!("getBooleanVectorAsList: {:?}", getBooleanVectorAsList);
callback_void("setBooleanVectorAsList", &[param(getBooleanVectorAsList)]);

println!("getBooleanMatrix");
let getBooleanMatrix: Vec<Vec<bool>> = callback("getBooleanMatrix", &[]);
println!("getBooleanMatrix: {:?}", getBooleanMatrix);
callback_void("setBooleanMatrix", &[param(getBooleanMatrix)]);

println!("getBooleanMatrixAsList");
let getBooleanMatrixAsList: Vec<Vec<bool>> = callback("getBooleanMatrixAsList", &[]);
println!("getBooleanMatrixAsList: {:?}", getBooleanMatrixAsList);
callback_void("setBooleanMatrixAsList", &[param(getBooleanMatrixAsList)]);

File: invesdwin-scripting-parent/invesdwin-scripting-haskell/invesdwin-scripting-haskell-runtime-contract/src/test/java/de/invesdwin/scripting/haskell/runtime/contract/callback/ParametersAndReturnsTestBoolean.hs
putStrLn ( show ( "getBoolean" ) )
getBoolean :: Bool <- callback "getBoolean" []
putStrLn ( show ( getBoolean ) )
callback "setBoolean" [param getBoolean] :: IO ()

putStrLn ( show ( "getBooleanVector" ) )
getBooleanVector :: [Bool] <- callback "getBooleanVector" []
putStrLn ( show ( getBooleanVector ) )
callback "setBooleanVector" [param getBooleanVector] :: IO ()

putStrLn ( show ( "getBooleanVectorAsList" ) )
getBooleanVectorAsList :: [Bool] <- callback "getBooleanVectorAsList" []
putStrLn ( show ( getBooleanVectorAsList ) )
callback "setBooleanVectorAsList" [param getBooleanVectorAsList] :: IO ()

putStrLn ( show ( "getBooleanMatrix" ) )
getBooleanMatrix :: [[Bool]] <- callback "getBooleanMatrix" []
putStrLn ( show ( getBooleanMatrix ) )
callback "setBooleanMatrix" [param getBooleanMatrix] :: IO ()

putStrLn ( show ( "getBooleanMatrixAsList" ) )
getBooleanMatrixAsList :: [[Bool]] <- callback "getBooleanMatrixAsList" []
putStrLn ( show ( getBooleanMatrixAsList ) )
callback "setBooleanMatrixAsList" [param getBooleanMatrixAsList] :: IO ()
